### DISCUSSION TITLE
Generate CLI from my API

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi/fastapi#4687] [Questions] [answered]
I managed to write my API with FastAPI. I can start it and under localhost:8080/docs I see all my routes and endpoints. Furthermore, there is a panel to authenticate with auth0. Also, for the remainder of this discussion we can assume that it is deployed at api.mydomain.com .

I would like to generate a CLI, with which I can authenticate (via auth0) and can call all my endpoints (perhaps even be able to set the endpoint domain at beginning like I wish).

The call should not involve anything like `python` or `npx`, but just look like 

```
mycoolCLI do this and that
```

Is there an easy and elegant way to do that when already having a running API with FastAPI? I mean in most respects it would just replace cURL requests ...

**Comment by BeeMoe5** [comment, marked as answer]
if you mean an http client, try [postman](https://www.postman.com/) or [insomnia](https://insomnia.rest/)

i should also mention pycharm pro has an [http client](https://www.jetbrains.com/help/pycharm/http-client-in-product-code-editor.html) built in, too. and also fastapi has a [test client](https://fastapi.tiangolo.com/tutorial/testing/?h=testi#using-testclient) too
